src/components/Incidents/IncidentsChart/IncidentsChart.jsx

```diff
@@ -6,103 +6,147 @@ import {
-              showGrid
-              tickFormat={(t) =>
-                new Date(t).toLocaleDateString('en-US', { month: 'short', day: 'numeric' })
+            <Chart
+              containerComponent={
+                <ChartVoronoiContainer
+                  labelComponent={
+                    <ChartTooltip constrainToVisibleArea labelComponent={<ChartLabel />} />
+                  }
+                  labels={({ datum }) =>
+                    `Severity: ${datum.name}\nComponent: ${datum.component}\nIncident ID: ${
+                      datum.group_id
+                    }\nStart: ${formatDate(new Date(datum.y0), true)}\nEnd: ${formatDate(
+                      new Date(datum.y),
+                      true,
+                    )}`
+                  }
+                />
               }
-              tickValues={dateValues}
-            />
-            <ChartGroup horizontal>
-              {chartData.map((bar, index) => {
-                return (
-                  //we have several arrays and for each array we make a ChartBar
-                  <ChartBar
-                    data={bar}
-                    key={index}
-                    style={{
-                      data: {
-                        fill: ({ datum }) => datum.fill,
-                        stroke: ({ datum }) => datum.fill,
-                      },
-                    }}
-                  />
-                );
-              })}
-            </ChartGroup>
-          </Chart>
-        </div>
-      )}
+              domainPadding={{ x: [30, 25] }}
+              legendData={[
+                { name: 'Critical', symbol: { fill: global_danger_color_100.var } },
+                { name: 'Info', symbol: { fill: global_info_color_100.var } },
+                { name: 'Warning', symbol: { fill: global_warning_color_100.var } },
+              ]}
+              legendPosition="bottom-left"
+              //this should be always less than the container height
+              height={500}
+              padding={{
+                bottom: 75, // Adjusted to accommodate legend
+                left: 50,
+                right: 25, // Adjusted to accommodate tooltip
+                top: 0,
+              }}
+              //TODO: WIDTH SHOULD BE AUTOMATICALLY ADJUSTED
+              width={width}
+            >
+              <ChartAxis
+                dependentAxis
+                showGrid
+                tickFormat={(t) =>
+                  new Date(t).toLocaleDateString('en-US', { month: 'short', day: 'numeric' })
+                }
+                tickValues={dateValues}
+              />
+              <ChartGroup horizontal>
+                {chartData.map((bar, index) => {
+                  return (
+                    //we have several arrays and for each array we make a ChartBar
+                    <ChartBar
+                      name={bar.group_id}
+                      data={bar}
+                      key={index}
```

Comment:
is there a better index we can use as a key?, this will prevent unnecessary re renders

Reply:
I am not using name prop, so I can replace index with bar.group_id